### PR TITLE
Fix getting delegate adapter for more types.

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java [hunk 1]
```diff
@@ -297,9 +297,9 @@ public void bind(Moshi moshi, JsonAdapter.Factory factory) {
           Type type = ((ParameterizedType) parameterTypes[i]).getActualTypeArguments()[0];
           Set<? extends Annotation> jsonAnnotations = jsonAnnotations(parameterAnnotations[i]);
           jsonAdapters[i - adaptersOffset] =
-              Types.equals(this.type, type) && annotations.equals(jsonAnnotations)
-                  ? moshi.nextAdapter(factory, type, jsonAnnotations)
-                  : moshi.adapter(type, jsonAnnotations);
+              Types.equals(Types.getRawType(this.type), Types.getRawType(type))
```
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java]: I think this will equate `List<String>` with `List<Integer>` ?

Comment (NightlyNexus) [inline moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java]: Oops, I believe I meant `canonicalize`.

File: moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java [hunk 2]
```diff
@@ -297,9 +297,9 @@ public void bind(Moshi moshi, JsonAdapter.Factory factory) {
           Type type = ((ParameterizedType) parameterTypes[i]).getActualTypeArguments()[0];
           Set<? extends Annotation> jsonAnnotations = jsonAnnotations(parameterAnnotations[i]);
           jsonAdapters[i - adaptersOffset] =
-              Types.equals(this.type, type) && annotations.equals(jsonAnnotations)
-                  ? moshi.nextAdapter(factory, type, jsonAnnotations)
-                  : moshi.adapter(type, jsonAnnotations);
+              Types.equals(Types.canonicalize(this.type), Types.canonicalize(type))
```
Comment (NightlyNexus) [inline moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java]: Yes, the test fails without it.
Cool, I'll send a PR to replace this one, then.

**Comment by NightlyNexus** [comment]
Fixed by https://github.com/square/moshi/pull/279!

**Pull request closed without merging**